app/src/main/java/com/android/streetworkapp/ui/park/ParkOverview.kt

```diff
@@ -40,33 +40,43 @@ import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
-import androidx.navigation.compose.rememberNavController
 import com.android.sample.R
 import com.android.streetworkapp.model.event.Event
 import com.android.streetworkapp.model.event.EventOverviewUiState
 import com.android.streetworkapp.model.event.EventViewModel
 import com.android.streetworkapp.model.park.Park
+import com.android.streetworkapp.model.park.ParkViewModel
 import com.android.streetworkapp.ui.navigation.NavigationActions
 import com.android.streetworkapp.ui.navigation.Screen
 import com.android.streetworkapp.utils.toFormattedString
 
 /**
  * Display the overview of a park, including park details and a list of events.
  *
- * @param park The park data to display.
+ * @param parkViewModel The view model for the park.
+ * @param innerPadding The padding to apply to the screen.
+ * @param navigationActions The navigation actions to navigate to other screens.
+ * @param eventViewModel The view model for the events.
  */
 @Composable
 fun ParkOverviewScreen(
-    park: Park,
+    parkViewModel: ParkViewModel,
     innerPadding: PaddingValues = PaddingValues(0.dp),
-    navigationActions: NavigationActions = NavigationActions(rememberNavController()),
+    navigationActions: NavigationActions,
     eventViewModel: EventViewModel
 ) {
+  val currentPark = parkViewModel.currentPark.collectAsState()
+
+  parkViewModel.park.collectAsState().value?.pid?.let { parkViewModel.getParkByPid(it) }
+
+  currentPark.value?.let { eventViewModel.getEvents(it) }
+
   Box(modifier = Modifier.padding(innerPadding).fillMaxSize().testTag("parkOverviewScreen")) {
     Column {
-      ImageTitle(image = null, title = park.name) // TODO: Fetch image from Firestore storage
-      ParkDetails(park = park)
-      EventItemList(eventViewModel) // TODO: Fetch events from Firestore
+      ImageTitle(image = null, title = currentPark.value?.name ?: "loading...")
+      // TODO: Fetch image from Firestore storage
+      currentPark.value?.let { ParkDetails(park = it) }
+      EventItemList(eventViewModel, navigationActions) // TODO: Fetch events from Firestore
```

Comment:
Question: why this is still a TODO since on my phone I do see the `events` when I pressed on a `park`?

Reply:
None honestly, i'll remove it

Reply:
Resolved